third_party/gfootball_engine/src/systems/graphics/rendering/opengl_renderer3d.cpp

```diff
@@ -1616,6 +1644,22 @@ void OpenGLRenderer3D::ResizeTexture(int textureID, SDL_Surface *source,
   DO_VALIDATION;
 
   BindTexture(textureID);
+  int x = 0;
+  int y = 0;
+  int width = source->w;
+  int height = source->h;
+
+  SDL_LockSurface(source);
+  mapping.glTexImage2D(GL_TEXTURE_2D, 0,
+          GetGLInternalPixelFormat(internalPixelFormat), width,
+          height, 0, GetGLPixelFormat(pixelFormat),
+          GL_UNSIGNED_BYTE, source->pixels);
+  SDL_UnlockSurface(source);
+
+  if (mipmaps) {
+    DO_VALIDATION;
+    mapping.glGenerateMipmap(GL_TEXTURE_2D);
```

Comment:
Mipmaps should be generated after `glTexImage2D` call: 
https://www.khronos.org/opengl/wiki/Common_Mistakes#Automatic_mipmap_generation